SimPEG/electromagnetics/frequency_domain/survey.py

```diff
@@ -16,13 +16,22 @@ def __init__(self, source_list, **kwargs):
         super(Survey, self).__init__(source_list, **kwargs)
 
         _frequency_dict = {}
+        _source_location = {}
+        _source_location_by_sounding = {}
         for src in self.source_list:
             if src.frequency not in _frequency_dict:
                 _frequency_dict[src.frequency] = []
             _frequency_dict[src.frequency] += [src]
+            if src.i_sounding not in _source_location:
+                _source_location[src.i_sounding] = []
+                _source_location_by_sounding[src.i_sounding] = []
+            _source_location[src.i_sounding] += [src]
+            _source_location_by_sounding[src.i_sounding] += [src.location]
```

Comment:
@sgkang : this seems to break the NSEM survey that inherits from FDEM survey (because NSEM has no source location)

`'PlanewaveXYPrimary' object has no attribute '_location'`

Reply:
PS: it can be solved by explicitly setting `location=None` for NSEM.